### PR TITLE
Task 5

### PR DESCRIPTION
**Pull request by Yuliya-soul**
valuation criteria
Reviewers should verify the lambda functions by invoking them through provided URLs.

1 - File serverless.yml contains configuration for importProductsFile function
3 - The importProductsFile lambda function returns a correct response which can be used to upload a file into the S3 bucket
4 - Frontend application is integrated with importProductsFile lambda
5 - The importFileParser lambda function is implemented and serverless.yml contains configuration for the lambda
Additional (optional) tasks
+1 - async/await is used in lambda functions
+1 - importProductsFile lambda is covered by unit tests (aws-sdk-mock can be used to mock S3 methods - https://www.npmjs.com/package/aws-sdk-mock) -1
+1 - At the end of the stream the lambda function should move the file from the uploaded folder into the parsed folder (move the file means that file should be copied into parsed folder, and then deleted from uploaded folder)
total=8-1=7
https://github.com/Yuliya-soul/aws-shop-fe/pull/3  Frontend application

### CODE REVIEW

File: src/import-service/importFileParser.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          Bucket: BUCKET,
+          Key: decodeURI(record.s3.object.key),
+        })
+          .createReadStream()
+          .pipe(csv())
+          .on("data", (data) => console.log("Parsed new data", data))
+          .on("end", () => resolve())
+          .on("error", (error) => reject(error));
+      });
+      await s3
```
Comment (ReshetovItsMe) [inline src/import-service/importFileParser.ts:21]: you can do it `on('end', .....)`, but ok

File: src/import-service/importProductsFile.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { errorResponse, successResponse } from "./src/utils/apiResponseBuilder";
+import AWS from "aws-sdk";
+
+export const importProductsFile = async (event) => {
+  try {
+    const s3 = new AWS.S3({ region: "us-east-1", signatureVersion: "v4" });
```
Comment (ReshetovItsMe) [inline src/import-service/importProductsFile.ts:6]: for our region, better to host it on eu-west-1

File: src/import-service/importFileParser.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import csv from "csv-parser";
+import AWS from "aws-sdk";
+
+export const importFileParser = async (event) => {
+  const BUCKET = "import-service-5";
+  const s3 = new AWS.S3({ region: "us-east-1" });
```
Comment (NikG77) [inline src/import-service/importFileParser.ts:6]: Recommend putting BUCKET in a constant and reusing it

File: src/import-service/importProductsFile.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    const catalogName = event.queryStringParameters.name;
+    const catalogPath = `uploaded/${catalogName}`;
+
+    const params = {
+      Bucket: "import-service-5",
+      Key: catalogPath,
```
Comment (NikG77) [inline src/import-service/importProductsFile.ts:12]: Recommend putting BUCKET in a constant and reusing it

File: src/import-service/serverless.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  runtime: nodejs12.x
+  stage: ${opt:stage, 'dev'}
+  region: us-east-1
+  lambdaHashingVersion: 20201221
+  iamRoleStatements:
+    - Effect: 'Allow'
+      Action:
+        - 's3:ListBucket'
+      Resource: 'arn:aws:s3:::import-service-5'
+    - Effect: "Allow"
```
Comment (NikG77) [inline src/import-service/serverless.yml:17]: Rreusing constant

**Pull request closed without merging**